go/rows.go

```diff
@@ -77,9 +81,61 @@ func NewRows(ctx context.Context, athenaAPI athenaiface.AthenaAPI, queryID strin
 	if err := r.fetchNextPage(nil); err != nil {
 		return nil, err
 	}
+
+	if r.ResultOutput.NextToken == nil || *r.ResultOutput.NextToken == "" {
+		return &r, nil
+	}
```

Comment:
An interesting detail is that I fetch that s3 file only if there are multiple pages. If all results are present on the first page, then we don't need to download the S3 file.

I fetch the first page every time because we need metadata information anyway (like types of columns etc.). And doing one extra request is not that problematic for large queries anyway.